.github/workflows/go-build-and-test.yml

```
@@ -0,0 +1,25 @@
+name: build-and-test
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
```

Bump this one to the latest version and also pin it to the commit hash:
```suggestion
    - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
```

---

Good call